src/platform/unittest/DataPathTest.cpp

```diff
@@ -455,6 +532,21 @@ struct CxPlatSocket {
         if (QUIC_SUCCEEDED(InitStatus)) {
             CxPlatSocketGetLocalAddress(Socket, &Route.LocalAddress);
             CxPlatSocketGetRemoteAddress(Socket, &Route.RemoteAddress);
+#ifdef QUIC_USE_RAW_DATAPATH
+            if (!QuicAddrIsWildCard(&Route.RemoteAddress)) {
+                //
+                // This is a connected socket and its route must be resolved
+                // to be able to send traffic.
+                //
+                InitStatus = CxPlatResolveRoute(Socket, &Route, 0, &Route, ResolveRouteComplete);
```

Comment:
Any idea why my ResolveRouteComplete callback being passed to CxPlatResolveRoute doesn't compile? I've tried it both with and without wrapping the function definition in extern "C". The function signature perfectly matches, but it still claims not to be able to "convert" the argument:

E:\msquic\src\platform\unittest\DataPathTest.cpp(542,1): error C2664: 'HRESULT CxPlatResolveRoute(CXPLAT_SOCKET *,CXPLAT_ROUTE *,uint8_t,void *,CXPLAT_ROUTE_RESOLUTION_CALLBACK_HANDLER)': cannot convert argument 5 fro
m 'void (__cdecl *)(void *,const uint8_t *,uint8_t,BOOLEAN)' to 'CXPLAT_ROUTE_RESOLUTION_CALLBACK_HANDLER' [E:\msquic\build\windows\x64_schannel\src\platform\unittest\msquicplatformtest.vcxproj]

Reply:
InitStatus = CxPlatResolveRoute(Socket, &Route, 0, &Route, ResolveRouteComplete);
=>
InitStatus = CxPlatResolveRoute(Socket, &Route, 0, &Route, (CXPLAT_ROUTE_RESOLUTION_CALLBACK_HANDLER)ResolveRouteComplete);

Reply:
It might be the calling convention. We might need to specify calling convention on the declaration of all the quic_datapath.h callbacks, and use it when we define the callbacks. Please don't just cast to make the error go away.

Reply:
Solved. Was a const issue.